exercises/danny_hindson/ex02_oo_basics/src/ShapeSorter.h

```diff
@@ -0,0 +1,26 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2019 ISIS Rutherford Appleton Laboratory UKRI,
+//     NScD Oak Ridge National Laboratory, European Spallation Source
+//     & Institut Laue - Langevin
+// SPDX - License - Identifier: GPL - 3.0 +
+
+#ifndef SHAPESORTER_H_
+#define SHAPESORTER_H_
+
+#include "Shape.h"
+#include <vector>
+
+class ShapeSorter {
+public:
+  ShapeSorter(std::vector<Shape*> v) : shapeVec(v) {};
+  virtual ~ShapeSorter() {};
+  void printShapes(std::string ShapeType) const;
```

Comment:
Good use of `const` on the member definition.

By default C++ copies its arguments. Most of the time this is not what we want and methods like this can be defined as
```c++
  void printShapes(const std::string &ShapeType) const;
```
so they are passed as a reference and will not copy the input the string.